### ISSUE TITLE
[BUG] gitlab: git shorthand broken in 7.7.0/7.7.1

### ISSUE THREAD

**Issue by feelepxyz** [npm/cli#2939]
### Current Behavior:
Running `npm install gitlab:kornelski/babel-preset-php` with npm@7.7.0/7.7.1 fails with:

```
Wrong spec type (git) for RemoteFetcher. Supported types: remote
```

### Expected Behavior:
Running `npm install gitlab:kornelski/babel-preset-php` with npm@7.7.0 + succeeds with:

```
added 4 packages, removed 1 package, and audited 5 packages in 4s
```

This works if you install 7.6.3: `npm install -g npm@7.6.3`

### Steps To Reproduce:
- `npm install -g npm@7.7.1`
- `npm install gitlab:kornelski/babel-preset-php`

### Environment:
- OS: MacOS Mojave
- Node: v15.6.0
- npm: 7.7.1

**Comment by nlf** [comment]
confirmed! fixed in hosted-git-info and will be in the next release coming out shortly

**Comment by nlf** [comment]
this should be fixed in 7.7.2, which was just published
